common/djangoapps/student/tests/test_models.py

```diff
@@ -713,6 +715,43 @@ def test_not_enrolled_student_is_enrolled(self):
         assert actual_student.is_active is True
         assert actual_cea.user == student
 
+    @ddt.data(False, True)
+    def test_auto_enrollment_if_course_enrollment_closed(self, feature_enabled):
+        """
+        Test the following scenarios
+
+        1. Invited students who register when enrollment is closed are not enrolled if
+        DISABLE_ALLOWED_ENROLLMENT_IF_ENROLLMENT_CLOSED is True
+
+        2. Invited students who register when enrollment is closed are enrolled if
+        DISABLE_ALLOWED_ENROLLMENT_IF_ENROLLMENT_CLOSED is False
+        """
+
+        def register_and_enroll_student():
+            student = self._set_up_invited_student(
+                course=self.course,
+                active=False,
+                enrolled=False
+            )
+            student.is_active = True
+            # trigger the post_save callback
+            student.save()
+            return CourseEnrollment.get_enrollment(student, self.course.id)
+
+        # Set enrollment end date to a past date so that enrollment is ended
+        enrollment_end = datetime.datetime.now(pytz.UTC) - datetime.timedelta(days=2)
+        course_overview = CourseOverviewFactory.create(id=self.course.id)
+        course_overview.enrollment_end = enrollment_end
```

Comment:
Nit: factories allow setting attributes on init:

```suggestion
        course_overview = CourseOverviewFactory.create(id=self.course.id, enrollment_end=enrollment_end)
```

We can also apply this to `lms/djangoapps/courseware/tests/test_access.py`.